### DISCUSSION TITLE
Decouple yt-dlp dependency and supply it as a binary

### DISCUSSION THREAD

**Discussion by codedealer** [exwm/yt_clipper#50] [General]
I'd like to float a question of decoupling the largest dependency, yt-dlp, and supplying it as a binary just like ffmpeg. Yt-dlp is updated quite frequently and sometimes those updates are critical for functionality of the library (i.e. when youtube changes their anti-bot measures). In this case simply updating a binary seems an easier option.

I ask this now because the latest update has bumped a minor version of python to 3.9 which would probably trigger some other dependency updates anyway so it might be a good time to consider an alternative, if it's viable for this project.

**Comment by exwm** [comment]
I've thought about it before, but never pulled the trigger on such a refactor. An alternative would be to automate upgrading of yt-dlp and building new versions of yt_clipper in CI.

When installing and running yt_clipper from source, e.g. with `poetry` or `pipx` it's fairly easy to change the yt-dlp version yourself so I would consider this mainly to be a problem for the binary releases.

I also looked at ways to add a feature to yt_clipper to upgrade the yt-dlp dependency to the latest. I'm not sure there is a straightforward way with pyinstaller, but maybe some other approach would work. If the yt-dlp binary was bundled with yt_clipper then this might be as easy as downloading and replacing the binary.

If someone wants to contribute one of these approaches to making it easier to update yt-dlp then I would accept a PR. I think I like having the yt-dlp version specified in the python environment, so it might be nice to keep that option available.

**Comment by exwm** [comment]
Completed in https://github.com/exwm/yt_clipper/releases/tag/v5.27.0
